checks.d/linuxvm.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+
+"""DataDog agent checks for system.vm proc info."""
+
+from checks import AgentCheck
+
+
+def valid_instance(instance):
+    return "proc" in instance or "vmstat" in instance
+
+
+class LinuxVm(AgentCheck):
+    VM_BASE = "system.vm"
+
+    def check(self, instance):
+        if not valid_instance(instance):
+            return
+
+        proc = instance.get("proc", {})
+        vmstat = instance.get("vmstat", {})
+
+        if proc:
+            for gauge in proc.get("gauges", []):
+                self.proc(self.gauge, gauge)
+
+            for counter in proc.get("counts", []):
+                self.proc(self.count, counter)
+
+        if vmstat:
+            vmstats = {x: int(y) for x, y in map(str.split, open("/proc/vmstat").readlines())}
```

Comment:
also it's good practice to always open files in a context manager:
```python
with open('/proc/vmstat') as f:
     content = vm.readlines()
```
this way you make sure the file handler will be properly closed.